docs/WALKTHROUGH.md

```diff
@@ -0,0 +1,394 @@
+
+These components are building blocks of the service catalog in Kubernetes from an
+API standpoint.
+
+----
+
+#### Note: accessing the service catalog
+
+Unfortunately, `kubectl` doesn't know how to speak to both the service catalog
+API server and the main Kubernetes API server without switching contexts or
+`kubeconfig` files.  For now, the best way to access the service catalog API
+server is via a dedicated `kubeconfig` file.  You can manage the kubeconfig in
+use within a directory using the `direnv` tool.
+
+### Creating kubeconfig
+Make sure to change to the service catalog directory and create the .kubeconfig
+file copying the following into your shell:
+
+console```
+cat << EOF > .kubeconfig
+apiVersion: v1
+clusters:
+- cluster:
+    server: http://<cluster ip>:80
+  name: service-catalog-cluster
+contexts:
+- context:
+    cluster: service-catalog-cluster
+  name: service-catalog-ctx
+current-context: service-catalog-ctx
+kind: Config
+preferences: {}
+EOF
+```
+
+The above `.kubeconfig` file used in combination with `direnv` is the simplest
+way to switch between the core API server and the service-catalog API server.
+An example of an .envrc file in the service catalog directory would look like:
+
+export KUBECONFIG=<path to service-catalog>/.kubeconfig
+
+Then kubectl commands executed from the service-catalog directory will
+now make requests to the correct API server.
+
+----
+
+Because we haven't created any resources in the service-catalog API server yet,
+`kubectl get` will return an empty list of resources:
+
+```console
+kubectl get brokers,serviceclasses,instances,bindings
+```
+
+Create a test-ns namespace while still executing Kubernetes API commands:
+```console
+kubectl create namespace test-ns
+```
+
+## Step 3 - Installing a UPS broker
+
+Service Catalog requires broker servers to operate properly. There is a User 
+Provided Service broker (UPS from now on), which allows consumption of existing
+services through the Service Catalog model. Just like any other broker, the
+UPS broker needs to be running somewhere before it can be added to the
+catalog. We need to deploy it first by using the
+[`ups-broker` Helm chart](../charts/ups-broker) into your cluster, just like
+you installed the catalog chart above. Run this simple command to install it with sensible
+defaults:
+
+```console
+helm install <path to service-catalog>/charts/ups-broker --name ups-broker --namespace ups-broker
+```
+
+## Step 4 - Creating a `Broker` Resource
+
+Next, we'll register a service broker with the catalog.  To do this, we'll
+create a new [`Broker`](../contrib/examples/walkthrough/ups-broker.yaml)
+resource against our API server.
+
+Before we do so, we'll need to configure our `kubeconfig` file as described above, and
+download a 1.6-beta version of `kubectl`. We'll need this version of `kubectl` for all 
+`create` operations against the service-catalog API server.
```

Comment:
the current dev guide says that users should make `kubectl` resolve to the 1.6 CLI for everything: https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/DEVGUIDE.md#note-accessing-the-service-catalog

can you change to that?